### PR TITLE
Delete `TrainingEpochLoop._dataloader_idx` which always equals 0

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

See title.

A follow-up could deprecate the argument for `on_train_batch_{start,end}`.

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -272,7 +272,7 @@ def logger(self):
         return self.trainer.logger if self.trainer else None
 
     def _apply_batch_transfer_handler(
-        self, batch: Any, device: Optional[torch.device] = None, dataloader_idx: Optional[int] = None
+        self, batch: Any, device: Optional[torch.device] = None, dataloader_idx: int = 0
```
Comment (awaelchli) [inline pytorch_lightning/core/lightning.py:275]: will require adjusting tests :)

**Review by awaelchli — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/loops/batch/training_batch_loop.py [hunk 2]
```diff
@@ -92,13 +91,13 @@ def run(self, batch: Any, batch_idx: int, dataloader_idx: int) -> AttributeDict:
             return AttributeDict(signal=-1)
 
         # hook
-        response = self.trainer.call_hook("on_train_batch_start", batch, batch_idx, dataloader_idx)
+        response = self.trainer.call_hook("on_train_batch_start", batch, batch_idx, 0)
```
Comment (rohitgr7) [inline pytorch_lightning/loops/batch/training_batch_loop.py:94]: just curious, if it's 0 all the time in this hook, what's the point of even keeping it??

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/loops/batch/training_batch_loop.py (hunk 2 above)
Comment (justusschock) [reply pytorch_lightning/loops/batch/training_batch_loop.py:94]: backwards compatibility :) This has been part of the api for quite some time now, so probably there are a lot of people expecting this.

@carmocca do we have plans to deprecate the argument at all?

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/loops/batch/training_batch_loop.py (hunk 2 above)
Comment (carmocca) [reply pytorch_lightning/loops/batch/training_batch_loop.py:94]: > do we have plans to deprecate the argument at all?

I guess so, do you have any reason not to?

**Pull request merged**